### PR TITLE
Improve http status all on http error middleware

### PR DESCRIPTION
**Pull request by Jgaldos**
Fixes [#3851](https://github.com/scrapy/scrapy/issues/3851)

on this spider 
```
from scrapy import Spider, Request


class HttpStatusSpider(Spider):
    name = "http_status"

    def start_requests(self):
        http_status = [200, 300, 400, 500]
        urls = [
            f"http://httpbin.org/status/{status}" for status in http_status
        ]
        for url in urls:
            yield Request(
                url=url,
                callback=self.parse,
                meta={
                    "handle_httpstatus_all": False
                }
            )

    def parse(self, response, **kwargs):
        print(f"response url: {response.url}")
```
previous the fix was printing all elements on the list, after the fix just print 200 status

### CODE REVIEW

File: scrapy/spidermiddlewares/httperror.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def process_spider_input(self, response, spider):
         if 200 <= response.status < 300:  # common case
             return
         meta = response.meta
-        if 'handle_httpstatus_all' in meta:
+        if 'handle_httpstatus_all' in meta and meta['handle_httpstatus_all']:
```
Comment (Gallaecio) [inline scrapy/spidermiddlewares/httperror.py:35]: :lipstick: 

```suggestion
        if meta.get('handle_httpstatus_all', False):
```

Comment (Jgaldos) [reply scrapy/spidermiddlewares/httperror.py]: Fixed

**Comment by eLRuLL** [comment]
@Jgaldos could you please improve the documentation too? as @kmike mentioned we could be more descriptive [here](https://docs.scrapy.org/en/latest/topics/spider-middleware.html#std:reqmeta-handle_httpstatus_all)

File: docs/topics/spider-middleware.rst [hunk 2]
```diff
@@ -255,7 +255,8 @@ this::
 The ``handle_httpstatus_list`` key of :attr:`Request.meta
 <scrapy.http.Request.meta>` can also be used to specify which response codes to
 allow on a per-request basis. You can also set the meta key ``handle_httpstatus_all``
-to ``True`` if you want to allow any response code for a request.
+to ``True`` if you want to allow any response code for a request, and to ``False``
+to disable it.
```
Comment (Gallaecio) [inline docs/topics/spider-middleware.rst:259]: I think ‘disable it’ here is slightly ambiguous. Maybe ‘disable the effects of the ``handle_httpstatus_list`` key’?

Comment (Jgaldos) [reply docs/topics/spider-middleware.rst:259]: I agree, I am changing it

File: docs/topics/spider-middleware.rst [hunk 3]
```diff
@@ -255,7 +255,8 @@ this::
 The ``handle_httpstatus_list`` key of :attr:`Request.meta
 <scrapy.http.Request.meta>` can also be used to specify which response codes to
 allow on a per-request basis. You can also set the meta key ``handle_httpstatus_all``
-to ``True`` if you want to allow any response code for a request.
+to ``True`` if you want to allow any response code for a request, and ``False`` to
+disable the effects of the ``handle_httpstatus_list`` key.
```
Comment (Gallaecio) [inline docs/topics/spider-middleware.rst:259]: My bad, sorry!

```suggestion
disable the effects of the ``handle_httpstatus_all`` key.
```

Comment (Jgaldos) [reply docs/topics/spider-middleware.rst:259]: fixed

**Review by Gallaecio — APPROVED** [review]

**Comment by eLRuLL** [comment]
@Gallaecio what should be the procedure to merge this? Are we waiting for another reviewer to accept and merge this? Please let me know if I can help in any way

**Comment by Gallaecio** [comment]
All reviews are more than welcome, but yes, at the moment we require 2 core developer approvals to merge.

**Pull request merged**